paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/EditPaymentMethodViewInteractor.kt

```diff
@@ -0,0 +1,169 @@
+package com.stripe.android.paymentsheet.ui
+
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.PaymentMethod
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+internal typealias PaymentMethodRemoveOperation = suspend (paymentMethod: PaymentMethod) -> Boolean
+internal typealias PaymentMethodUpdateOperation = suspend (
+    paymentMethod: PaymentMethod,
+    brand: CardBrand
+) -> Result<PaymentMethod>
+
+internal interface EditPaymentMethodViewInteractor {
+    val viewState: StateFlow<EditPaymentMethodViewState>
+
+    fun handleViewAction(viewAction: EditPaymentMethodViewAction)
+}
+
+internal interface ModifiableEditPaymentMethodViewInteractor : EditPaymentMethodViewInteractor {
+    fun close()
+
+    interface Factory {
+        fun create(
+            initialPaymentMethod: PaymentMethod,
+            onRemove: PaymentMethodRemoveOperation,
+            onUpdate: PaymentMethodUpdateOperation
+        ): ModifiableEditPaymentMethodViewInteractor
+    }
+}
+
+internal class DefaultEditPaymentMethodViewInteractor constructor(
+    initialPaymentMethod: PaymentMethod,
+    private val onRemove: PaymentMethodRemoveOperation,
+    private val onUpdate: PaymentMethodUpdateOperation,
```

Comment:
Should these be called `removeExecutor` and `updateExecutor`? The purpose is to do the actual work, but the name doesn’t suggest that.